maintenance/tracing/tracing.go

```diff
@@ -31,8 +31,7 @@ func init() {
 		return
 	}
 	if cfg.ServiceName == "" {
-		log.Warn("Using Jaeger noop tracer since no JAEGER_SERVICE_NAME is present")
-		return
+		log.Fatal("JAEGER_SERVICE_NAME is not present")
```

Comment:
The advantage is: we get a non-zero exit code of the program now?

Reply:
I would not change the behaviour to fatal since this is called in an init function what can have unexpected behaviour changes at other places. If a missing `ServiceName ` results in a program failure, we should instead make the original warning a valid case ("Using noop ...")

Reply:
Then let's restructure it completely and dump the code in init() and move it somewhere explicit?